app/models/mixins/reserved_mixin.rb

```diff
@@ -1,30 +1,23 @@
 module ReservedMixin
   extend ActiveSupport::Concern
   included do
-    has_one :reserved_rec, :class_name => "::Reserve", :as => :resource, :dependent => :delete
-
-    # Override the method used by save and save!
-    def create_or_update_with_reserved
-      # Touch the current record, but don't save it yet
-      current_time = current_time_from_proper_timezone
-      write_attribute('updated_at', current_time) if respond_to?(:updated_at)
-      write_attribute('updated_on', current_time) if respond_to?(:updated_on)
-
-      ret = create_or_update_without_reserved
-      res = reserved_rec
-      res.save! if res
-      ret
-    end
-    alias_method_chain :create_or_update, :reserved
+    has_one :reserved_rec, :class_name => "::Reserve", :as => :resource,
+      :autosave => true, :dependent => :delete
```

Comment:
I vaguely remember a specific reason I didn't use autosave when I wrote this nastiness.  I'd have to check git blame, but I just wanted to do a quick comment here to remind myself.